scheduler/test/tbf/test/ratelimit.clj

```diff
@@ -0,0 +1,92 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns tbf.test.ratelimit
+  (:use clojure.test)
+  (:require [tbf.core]
+            [tbf.ratelimit]))
+
+(deftest independent-keys-1
+  (let [ratelimit (tbf.ratelimit/tbf-ratelimiter-make 60000 60 10)]
+    (tbf.ratelimit/earn-tokens! ratelimit "Foo1")
+    (tbf.ratelimit/time-until ratelimit "Foo2")
+    (tbf.ratelimit/in-debt? ratelimit "Foo3")
+    (tbf.ratelimit/spend-tokens! ratelimit "Foo4" 100)
+    (is (= 4 (.size (.asMap (:cache ratelimit)))))))
+
+(deftest independent-keys-2
```

Comment:
imho, there should be an assertion on the state of the tbf after each of the api calls in this test.

Reply:
Fixed.